hub/gads-ui/src/components/DeviceControl/StreamCanvas/StreamCanvas.js

```diff
@@ -39,43 +48,242 @@ export default function StreamCanvas({ deviceData, shouldShowStream }) {
 
     const handleOrientationButtonClick = (isPortrait) => {
         setIsPortrait(isPortrait)
+        updateCanvasDimensions(isPortrait)
     }
 
+    // Handles orientation/resizing only
+    useEffect(() => {
+        const handleResize = () => {
+            // Only update if the ref still exists
+            if (videoDimensionsRef.current) {
+                updateCanvasDimensions(isPortrait)
+            }
+        }
+
+        updateCanvasDimensions(isPortrait)
+
+        window.addEventListener('resize', handleResize)
+        return () => window.removeEventListener('resize', handleResize)
+    }, [])
+
+    // Handles starting/stopping the WebRTC connection
     useEffect(() => {
-        const updateCanvasDimensions = () => {
-            let calculatedWidth, calculatedHeight
-            if (isPortrait) {
-                calculatedHeight = window.innerHeight * 0.7
-                calculatedWidth = calculatedHeight * deviceScreenRatio
+        if (!useWebRTCVideo) {
+            const imgElement = document.getElementById('image-stream')
+            // If MJPEG, just set the <img> src once
+            if (shouldShowStream) {
+                imgElement.src = streamUrl
             } else {
-                calculatedWidth = window.innerWidth * 0.4
-                calculatedHeight = calculatedWidth / deviceLandscapeScreenRatio
+                imgElement.src = ''
             }
+            return;
+        }
 
-            setCanvasDimensions({
-                width: calculatedWidth,
-                height: calculatedHeight
-            })
+        if (shouldShowStream) {
+            setupWebRTCVideo()
         }
 
-        const imgElement = document.getElementById('image-stream')
+        return () => {
+            // Only tear down if user hides the stream or unmounts
+            if (ws.current) ws.current.close()
+            if (pc.current) pc.current.close()
```

Comment:
Consider setting ws.current = null and pc.current = null after closing them to avoid potential reconnection issues.

Reply:
Done